src/rt/lifetime.d

```diff
@@ -437,17 +437,15 @@ BlkInfo __arrayAlloc(size_t arrsize, const TypeInfo ti, const TypeInfo tinext) n
 {
     import core.checkedint;
 
-    size_t typeInfoSize = structTypeInfoSize(tinext);
-    size_t padsize = arrsize > MAXMEDSIZE ? LARGEPAD : ((arrsize > MAXSMALLSIZE ? MEDPAD : SMALLPAD) + typeInfoSize);
-
     bool overflow;
-    auto padded_size = addu(arrsize, padsize, overflow);
+    immutable padsize = __arrayPad(arrsize, tinext);
+    immutable padded_size = addu(arrsize, padsize, overflow);
 
     if (overflow)
         return BlkInfo();
 
     uint attr = (!(tinext.flags & 1) ? BlkAttr.NO_SCAN : 0) | BlkAttr.APPENDABLE;
-    if (typeInfoSize)
+    if (structTypeInfoSize(tinext))
```

Comment:
The call to __arrayPad was inlined before to avoid calling structTypeInfoSize twice. Not sure if optimizers inline it and detect the duplication.

Reply:
I checked if `structTypeInfoSize` had any side effect, but it's only checking if the struct has destructors. I guess I would need to refactor this further, but it's not really worth it ATM, so reverting that change.